src/main/java/ro/unibuc/hello/service/DoneeService.java

```diff
@@ -0,0 +1,42 @@
+package ro.unibuc.hello.service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.bson.types.ObjectId;
+import org.springframework.beans.BeanUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import ro.unibuc.hello.data.CharityEventRepository;
+import ro.unibuc.hello.data.DoneeEntity;
+import ro.unibuc.hello.dto.DoneeDTO;
+
+@Component
+public class DoneeService {
+  @Autowired
+  private CharityEventRepository charityEventRepository;
+  
+  public void addDoneesToCharity (String charityId, List<DoneeDTO> donees) {
+    var charity = charityEventRepository.findById(charityId);
+    if (charity.isPresent()) {
```

Comment:
You can replace this with
```suggestion
    if (!charity.isPresent()) {
      return;
    }
```
to reduce nesting.

And my IDE is warning me that `!charity.isPresent()` is the same as `charity.isEmpty()`, so you can use that.